custom/icds_reports/static/icds_reports/js/spec/kpi.directive.spec.js

```diff
@@ -0,0 +1,93 @@
+/* global module, inject */
+"use strict";
+
+var pageData = hqImport('hqwebapp/js/initial_page_data');
+
+
+describe('KpiDirective', function () {
+
+    var $scope, $compile, $location, controller, $httpBackend;
+
+    pageData.registerUrl('icds-ng-template', 'template');
+    pageData.registerUrl('icds_locations', 'icds_locations');
+
+    beforeEach(module('icdsApp'));
+
+    beforeEach(inject(function ($rootScope, _$compile_, _$location_, _$httpBackend_) {
+        $compile = _$compile_;
+        $httpBackend = _$httpBackend_;
+        $scope = $rootScope.$new();
+        $location = _$location_;
+
+        $httpBackend.expectGET('template').respond(200, '<div></div>');
+        var element = window.angular.element("<kpi></kpi>");
+
+        var compiled = $compile(element)($scope);
+
+        $httpBackend.flush();
+        $scope.$digest();
+        controller = compiled.controller("kpi");
+    }));
+
+    it('tests get empty page path', function () {
+        var result = controller.goToStep("test");
+        var expected = "#/test";
+        assert.equal(result, expected);
+    });
+
+    it('tests get page path with one arguments', function () {
+        $location.search('one', 1);
+        var result = controller.goToStep("test");
+
+        var expected = "#/test?one=1&";
+        assert.equal(result, expected);
+    });
+
+    it('tests get page path with two arguments', function () {
+        $location.search('one', 1);
+        $location.search('two', 2);
+        var result = controller.goToStep("test");
+
+        var expected = "#/test?one=1&two=2&";
+        assert.equal(result, expected);
+    });
+
+    it('tests not shows percent info', function () {
+        var expected = false;
+        var result = controller.showPercentInfo();
+        assert.equal(result, expected);
+    });
+
+    it('tests shows percent info from month parameter', function () {
+        $location.search('month', new Date().getMonth());
+        var expected = true;
+
+        var result = controller.showPercentInfo();
+        assert.equal(result, expected);
+    });
```

Comment:
This test now fails, even on the old commit which introduced this.  My guess is it depends on the current year or month in some way.
```
  1) Kpi Directive tests shows percent info from month parameter:
     AssertionError: expected false to equal true
      at http://localhost:8000/static/chai/chai.js:210
      at http://localhost:8000/static/chai/chai.js:2178
      at http://localhost:8000/static/icds_reports/js/spec/kpi.directive.spec.js:70
```
@pstudzinski @lwyszomi  could you please investigate?

I've removed it in the meantime: https://github.com/dimagi/commcare-hq/pull/18998

Reply:
Thanks. Sure I fixed it.
PR: https://github.com/dimagi/commcare-hq/pull/19010
@esoergel @lwyszomi